### ISSUE TITLE
use VSCode's snippet variables for some snippets

### ISSUE THREAD

**Issue by selrond** [dsznajder/vscode-es7-javascript-react-snippets#20]
[Snippet variables](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_variables)

for example

```json
  "reactFunctionalExportComponent": {
    "prefix": "rfe",
    "body": [
      "import React from 'react'",
      "",
      "const ${1:componentName} = () => {",
      "  return (",
      "    <div>",
      "      $0",
      "    </div>",
      "  )",
      "}",
      "",
      "export default ${1:componentName}",
      ""
    ],
    "description": "Creates a React Functional Component with ES7 module system"
  },
```

instead of `componentName` default string, there could be `TM_FILENAME_BASE` variable, to automatically name the component after the file name.

Saves time, it's more useful, while still stays selected, though easily rewritable.

Win-win.

**Comment by selrond** [comment]
@dsznajder 👌 THX

**Comment by selrond** [comment]
@dsznajder how about `exp→` `export default moduleName` snippet? It would make sense to use filename base as the default as well

**Comment by selrond** [comment]
and btw, `rfe` snippet is expanding to 

```jsx
import React from 'react'

const TM_FILENAME_BASE = () => {
	return (
		<div>
			
		</div>
	)
}

export default TM_FILENAME_BASE
```

you should use it like `${1:${TM_FILENAME_BASE}}`

**Comment by dsznajder** [comment]
@selrond Oh, I failed with implementation xD. I will fix this tomorrow morning, Documentation for this is little unclear :/

**Comment by dsznajder** [comment]
Will be released in v1.8.3 :)
